dask/array/tests/test_cupy.py

```diff
@@ -78,3 +103,490 @@ def test_sizeof(dtype):
+    assert e.shape[0] == d.shape[0] + 4
+    assert e.shape[1] == d.shape[1]
+
+    assert_eq(e[1, :], d[-1, :])
+    assert_eq(e[0, :], d[-2, :])
+
+
+@pytest.mark.skipif(missing_arrfunc_cond, reason=missing_arrfunc_reason)
+def test_reflect():
+    x = cupy.arange(10)
+    d = da.from_array(x, chunks=(5, 5), asarray=False)
+
+    e = da.overlap.reflect(d, axis=0, depth=2)
+    expected = np.array([1, 0, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 9, 8])
+    assert_eq(e, expected)
+
+    e = da.overlap.reflect(d, axis=0, depth=1)
+    expected = np.array([0, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 9])
+    assert_eq(e, expected)
+
+
+@pytest.mark.skipif(missing_arrfunc_cond, reason=missing_arrfunc_reason)
+def test_nearest():
+    x = cupy.arange(10)
+    d = da.from_array(x, chunks=(5, 5), asarray=False)
+
+    e = da.overlap.nearest(d, axis=0, depth=2)
+    expected = np.array([0, 0, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 9, 9])
+    assert_eq(e, expected)
+
+    e = da.overlap.nearest(d, axis=0, depth=1)
+    expected = np.array([0, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 9])
+    assert_eq(e, expected)
+
+
+@pytest.mark.xfail(reason="no shape argument support *_like functions on CuPy yet")
+@pytest.mark.skipif(np.__version__ < '1.17', reason='no shape argument for *_like functions')
+@pytest.mark.skipif(missing_arrfunc_cond, reason=missing_arrfunc_reason)
+def test_constant():
+    x = cupy.arange(64).reshape((8, 8))
+    d = da.from_array(x, chunks=(4, 4), asarray=False)
+
+    e = da.overlap.constant(d, axis=0, depth=2, value=10)
+    assert e.shape[0] == d.shape[0] + 4
+    assert e.shape[1] == d.shape[1]
+
+    assert_eq(e[1, :], np.ones(8, dtype=x.dtype) * 10)
+    assert_eq(e[-1, :], np.ones(8, dtype=x.dtype) * 10)
+
+
+@pytest.mark.xfail(reason="no shape argument support *_like functions on CuPy yet")
+@pytest.mark.skipif(np.__version__ < '1.17', reason='no shape argument for *_like functions')
+@pytest.mark.skipif(missing_arrfunc_cond, reason=missing_arrfunc_reason)
+def test_boundaries():
+    x = cupy.arange(64).reshape((8, 8))
+    d = da.from_array(x, chunks=(4, 4), asarray=False)
+
+    e = da.overlap.boundaries(d, {0: 2, 1: 1}, {0: 0, 1: 'periodic'})
+
+    expected = np.array(
+        [[ 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+         [ 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+         [ 7, 0, 1, 2, 3, 4, 5, 6, 7, 0],
+         [15, 8, 9,10,11,12,13,14,15, 8],
+         [23,16,17,18,19,20,21,22,23,16],
+         [31,24,25,26,27,28,29,30,31,24],
+         [39,32,33,34,35,36,37,38,39,32],
+         [47,40,41,42,43,44,45,46,47,40],
+         [55,48,49,50,51,52,53,54,55,48],
+         [63,56,57,58,59,60,61,62,63,56],
+         [ 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+         [ 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]])
+    assert_eq(e, expected)
+
+
+def test_random_all():
+    def rnd_test(func, *args, **kwargs):
+        a = func(*args, **kwargs)
+        assert type(a._meta) == cupy.core.core.ndarray
+        assert type(a.compute()) == cupy.core.core.ndarray
```

Comment:
Yes, I feared that it would look ridiculous and someone would end up removing that at one point. I'm going to do that and add a comment that it's done on purpose.

Reply:
We use `assert_eq(a, a)` in other places as well.  I recommend just doing that.  I think that most of the maintainers know that this is common, just because of all of the other checks that `assert_eq` runs.